rpc/server.go

```diff
@@ -358,14 +475,45 @@ func (s *Server) exec(ctx context.Context, codec ServerCodec, req *serverRequest
 // execBatch executes the given requests and writes the result back using the codec.
 // It will only write the response back when the last request is processed.
 func (s *Server) execBatch(ctx context.Context, codec ServerCodec, requests []*serverRequest) {
+	lctx := ctx
+	s.pendingMu.Lock()
+	codecPending := s.pending[codec]
+	var reqIDs []interface{}
+	var preq *pendingRequest
+	if codecPending != nil {
+		reqIDs = make([]interface{}, len(requests))
+		var cancelFn func()
+		lctx, cancelFn = context.WithCancel(ctx)
+		preq = &pendingRequest{cancelFn, false}
+		for i, req := range requests {
+			reqIDs[i], _ = parseReqId(*req.id.(*json.RawMessage))
```

Comment:
On second thought, that would still be json-specific behavior. Maybe the json codec should properly decode the request ID so that its dynamic type is either float64 or string, not *json.RawMessage